servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/ReplayStrategies.java

```diff
@@ -158,23 +157,29 @@ private static final class LazyTimeLimitedReplayAccumulator<T> implements Replay
 
         @Override
         public void accumulate(@Nullable final T t) {
+            final long nanoTime = executor.currentTime(NANOSECONDS);
+            trimExpired(nanoTime);
```

Comment:
this incurs an iteration of all history for each `onNext` which makes costs proportional to history size. is it worth doing this since we commit to the memory anyways? they can always use the eager strategy if they want more aggressive memory reclamation.